api/common/logs.go

```diff
@@ -116,6 +116,7 @@ func StreamDebugLog(ctx context.Context, source base.StreamConnector, args Debug
 
 	messages := make(chan LogMessage)
 	go func() {
+		defer func() { _ = connection.Close() }()
```

Comment:
Should we be adding a metrics increment of another connection?

Reply:
I am not sure how to do that, since we are in api/common.

This code is called in both:

* [migrationmaster/client.go](https://github.com/juju/juju/blob/fc069f19f894c4f7c777ed9a290cd64a706c24dd/api/controller/migrationmaster/client.go#L327) for [transferring logs](https://github.com/juju/juju/blob/7130b5b03d0f7209adfbd8831156396f7dc7df85/worker/migrationmaster/worker.go#L615)
* [api/client](https://github.com/juju/juju/blob/76a59def09b1acc73289ff7b420bc8432d48c9fe/api/client/client/client.go#L237), but used only by juju command [debug-log](https://github.com/juju/juju/blob/76a59def09b1acc73289ff7b420bc8432d48c9fe/cmd/juju/commands/debuglog.go#L284)

Both seems pretty occasional, so even if there was actually a leak, it is not the one which causes the observed behavior.

Moreover, maybe I am wrong, but I don't see any metrics for connection on client side (such kind of metrics are on server side, as far I've dug)